src/ChronosTime.php

```diff
@@ -143,11 +188,11 @@ public function getSeconds(): int
      */
     public function setSeconds(int $seconds): static
     {
-        $prevSecondsTicks = $this->ticks % self::TICKS_PER_MINUTE - $this->ticks % self::TICKS_PER_SECOND;
-        $newSecondsTicks = static::mod($seconds, 60) * self::TICKS_PER_SECOND;
+        $baseTicks = $this->ticks - ($this->ticks % self::TICKS_PER_MINUTE - $this->ticks % self::TICKS_PER_SECOND);
```

Comment:
Same calculation as before but we subtract from the total ticks here and call it base instead of doing it below.